### ISSUE TITLE
Model files saved are quite bigger than the provided model file

### ISSUE THREAD

**Issue by eypros** [DrSleep/tensorflow-deeplab-resnet#141]
I am trying to use deeplab-resnet on custom data and more precisely I am using the small dataset of Kitty you mentioned in the instructions.

I have noticed that every time I save a model's weights using fine_tune the resulting file is larger than the original provided by you. The problem is that if I use the train module the result is even bigger (I got files of 340mb) while when I changed the optimizer from your custom to AdamOptimizer (the one you use on fine_tune) I got a model file of 509mb!

Is there an explanation for that? I have't really changed any functional part of your code (just some sugar snippets) but it seems it appends tensors and operations without reason.

**Comment by DrSleep** [comment]
usually, the optimiser maintains additional values: e.g., previous gradient updates for SGD with momentum, or running first- and second-order statistics for Adam. That is why in practice the model size can be the double or even the triple of the original size.
